### PR TITLE
[cli] Topo: Migrate zk2topo and k8stopo to pflag

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
## Description
Migrate zk2topo and k8stopo flags to pflag

## Related Issue(s)
Fixes https://github.com/vitessio/vitess/issues/10860
Fixes https://github.com/vitessio/vitess/issues/10862

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/servenv/servenv.go [hunk 1]
```diff
@@ -385,3 +385,9 @@ func init() {
 	// Flags in package logutil are installed for all binaries.
 	OnParse(logutil.RegisterFlags)
 }
+
+func RegisterFlagsForTopoBinaries(registerFlags func(fs *pflag.FlagSet)) {
```
Comment (ajm188) [inline go/vt/servenv/servenv.go:389]: ugh, that's annoying, sorry about that!

**Review by ajm188 — COMMENTED** [review]
code changes looking good, whenever you're ready to move out of draft

**Comment by ajm188** [comment]
i don't think this has anything to do with flags, it's just environment variables: https://github.com/kubernetes/client-go/blob/e6538dd42b4fe55b6c754e41c66b43133ba41a59/rest/config.go#L516-L519

**Comment by ajm188** [comment]
ahhh, https://github.com/kubernetes/client-go/blob/e6538dd42b4fe55b6c754e41c66b43133ba41a59/tools/clientcmd/client_config.go#L614-L617 let's see where those come from

**Comment by ajm188** [comment]
https://github.com/vitessio/vitess/blob/main/go/vt/topo/k8stopo/server_flaky_test.go#L86-L92 cc @rohit-nayak-ps

**Comment by rohit-nayak-ps** [comment]
> i don't think this has anything to do with flags, it's just environment variables: https://github.com/kubernetes/client-go/blob/e6538dd42b4fe55b6c754e41c66b43133ba41a59/rest/config.go#L516-L519

Those environment variables are accessed from within a pod, which I believe, are passed via a kube config file and that seems to get messed up if we replace `flag` with `pflag` here: https://github.com/vitessio/vitess/blob/main/go/vt/topo/k8stopo/server.go#L50-L59

**Comment by ajm188** [comment]
> Those environment variables are accessed from within a pod, which I believe, are passed via a kube config file and that seems to get messed up if we replace flag with pflag here:

do you know where that config file lives? maybe they're using single dashes?

i don't understand how this couldn't work, unless that pod is starting up via a binary that doesn't use `servenv` for parsing

**Comment by deepthi** [comment]
A few more help text files need to be fixed
```
$ find go/flags/endtoend -iname '*.txt' | xargs basename | cut -d. -f1 | xargs -I{} bash -c "{} --help 2>go/flags/endtoend/{}.txt"
```
and revert any changes to vtctldclient.txt

File: go/flags/endtoend/vtorc.txt [hunk 2]
```diff
@@ -59,15 +59,6 @@ Usage of vtorc:
...
-      --topo_k8s_namespace string                    The kubernetes namespace to use for all objects. Default comes from the context or in-cluster config
-      --topo_zk_auth_file string                     auth to use when connecting to the zk topo server, file contents should be <scheme>:<auth>, e.g., digest:user:pass
-      --topo_zk_base_timeout duration                zk base timeout (see zk.Connect) (default 30s)
-      --topo_zk_max_concurrency int                  maximum number of pending requests to send to a Zookeeper server. (default 64)
-      --topo_zk_tls_ca string                        the server ca to use to validate servers when connecting to the zk topo server
-      --topo_zk_tls_cert string                      the cert to use to connect to the zk topo server, requires topo_zk_tls_key, enables TLS
-      --topo_zk_tls_key string                       the key to use to connect to the zk topo server, enables TLS
```
Comment (deepthi) [inline go/flags/endtoend/vtorc.txt:70]: not sure about the other flags, but I'm pretty sure we want to keep these for vtorc.
cc @GuptaManan100

Comment (GuptaManan100) [reply go/flags/endtoend/vtorc.txt:70]: Yes, we do, whatever flags we need to connect to a topo-server. We'll need them.

Comment (rohit-nayak-ps) [reply go/flags/endtoend/vtorc.txt]: I updated the list of  topo binaries in servenv to add `vtorc`. I had just refactored the list from main into this, so I guess previous PRs too had this issue. 

```
func RegisterFlagsForTopoBinaries(registerFlags func(fs *pflag.FlagSet)) {
	topoBinaries := []string{
		"vtbackup",
		"vtcombo",
		"vtctl",
		"vtctld",
		"vtgate",
		"vtgr",
		"vttablet",
		"vttestserver",
		"zk",
		"vtorc",
	}
	for _, cmd := range topoBinaries {
		OnParseFor(cmd, registerFlags)
	}
}
```

File: go/vt/topo/k8stopo/server.go [hunk 3]
```diff
@@ -47,16 +48,24 @@ import (
...
 )
 
-var (
+var kubeconfigPath, configContext, configNamespace string
+var k8stopoFlags *pflag.FlagSet
+
+func init() {
+	k8stopoFlags = pflag.NewFlagSet("k8stopo", pflag.ExitOnError)
+	registerFlags(k8stopoFlags)
+}
```
Comment (GuptaManan100) [inline go/vt/topo/k8stopo/server.go:57]: I don't understand this part 😕. Shouldn't we register the k8stopo like the other topo services?

Comment (rohit-nayak-ps) [reply go/vt/topo/k8stopo/server.go:57]: The other mechanisms didn't work. Maybe there is a better way, but I needed to do this in `go/vt/topo/k8stopo/server_flaky_test.go`  using the above technique  to fix `TestKubernetesTopo`.

```
k8stopoFlags.Set("topo_k8s_kubeconfig", testConfigPath)
```

Comment (GuptaManan100) [reply go/vt/topo/k8stopo/server.go:57]: Can't you directly set `kubeconfigPath = testConfigPath`? I think we do that in other test code too.
We use something like 
```go
oldConfigPath := kubeconfigPath
kubeconfigPath = testPath
defer func() {
    kubeconfigPath = oldConfigPath
}()
```

Comment (GuptaManan100) [reply go/vt/topo/k8stopo/server.go:57]: Also, don't we need these flags in the binaries that also have the zookeeper flags, like VTOrc etc?

Comment (GuptaManan100) [reply go/vt/topo/k8stopo/server.go:57]: I think they will be used if someone uses `k8s` flavour of topo-server, but I am not sure.

Comment (ajm188) [reply go/vt/topo/k8stopo/server.go:57]: imo, this code should happen only in the flaky test file (ie. you should do what things like `tabletconntest.SetProtocol` do)

File: go/flags/endtoend/vtorc.txt (hunk 2 above)
Comment (deepthi) [reply go/flags/endtoend/vtorc.txt]: `zk` doesn't need to be in this list, does it?

Comment (rohit-nayak-ps) [reply go/flags/endtoend/vtorc.txt]: That list comes from https://github.com/vitessio/vitess/pull/11106/files

File: go/vt/topo/k8stopo/server.go (hunk 3 above)
Comment (rohit-nayak-ps) [reply go/vt/topo/k8stopo/server.go]: @GuptaManan100's suggestion above worked.

File: go/vt/topo/k8stopo/server.go [hunk 4]
```diff
@@ -28,12 +28,15 @@ We follow these conventions within this package:
 package k8stopo
 
 import (
-	"flag"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"vitess.io/vitess/go/vt/servenv"
```
Comment (rsajwani) [inline go/vt/topo/k8stopo/server.go:36]: can you move this to down at line 46 with other vitess.io packages

**Review by rsajwani — APPROVED** [review]

**Pull request merged**